### PR TITLE
Fix test apps and clean up auth

### PR DESCRIPTION
**Pull request by calebmshafer**
This PR fixes the Electron sign-in workflow and updates both the ui-test-app and display-test-app to properly work when requiring a sign-in.

In addition to the above:

- Added the `x-correlation-id` header to every call to request if one isn't provided
- Removed the hardcoded client configurations from ui-test-app and display-test-app. Added new information to the READMEs to aid in getting set up.

### CODE REVIEW

**Comment by NancyMcCallB** [comment]
I pulled the latest changes in this branch, and I get the following when I try to start the app:
Unhandled Rejection (TypeError): Failed to construct 'URL': Invalid URL

If it helps, the full-stack-tests failed to build with the following error:
    src/backend/RpcImpl.ts(31,57): error TS2339: Property 'getUrl' does not exist on type 'IModelClient'.

**Comment by calebmshafer** [comment]
> If it helps, the full-stack-tests failed to build with the following error: src/backend/RpcImpl.ts(31,57): error TS2339: Property 'getUrl' does not exist on type 'IModelClient'.

@NancyMcCallB fixed the build issue.

> I pulled the latest changes in this branch, and I get the following when I try to start the app: Unhandled Rejection (TypeError): Failed to construct 'URL': Invalid URL

What way were you trying to run the app? Electron or web and debugging or non-debug?

**Comment by NancyMcCallB** [comment]
@calebmshafer I was running from the VSCode ui-test-app Chrome configuration. I think @bsteinbk had the same results running in the browser (chrome or firefox?) outside VSCode.

**Comment by calebmshafer** [comment]
@NancyMcCallB @bsteinbk mind going through the README for the setup steps and let me know if you see any issues?

@NancyMcCallB I'm trying to reproduce now...

File: clients/itwin/src/Request.ts [hunk 1]
```diff
@@ -284,6 +282,8 @@ export async function request(url: string, options: RequestOptions): Promise<Res
   if (options.headers)
     sareq = sareq.set(options.headers);
 
+  sareq = sareq.set(requestIdHeaderName, Guid.createValue());
```
Comment (calebmshafer) [inline clients/itwin/src/Request.ts:285]: TODO: Need to fix this so it only sets the header if it doesn't already have one.

File: core/frontend/src/tile/RealityModelTileTree.ts [hunk 2]
```diff
@@ -887,10 +887,14 @@ export class RealityModelTileClient {
...
 
   private async _doRequest(url: string, responseType: string): Promise<any> {
+    const authToken = this._requestAuthorization ?? (await IModelApp.getAccessToken());
+
     const options: RequestOptions = {
       method: "GET",
       responseType,
-      headers: this._requestAuthorization ? { authorization: this._requestAuthorization } : undefined,
+      headers: {
+        authroization: authToken,
```
Comment (mgooding) [inline core/frontend/src/tile/RealityModelTileTree.ts:896]: typo?

Comment (calebmshafer) [reply core/frontend/src/tile/RealityModelTileTree.ts]: thanks, fixed.

**Review by pmconne — APPROVED** [review]
`dta signin` is fixed and CORS errors when requesting reality data are resolved. Thanks!

File: clients/itwin/src/Request.ts (hunk 1 above)
Comment (calebmshafer) [reply clients/itwin/src/Request.ts]: fixed.

File: test-apps/ui-test-app/README.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+* IMJS_UITESTAPP_USE_LOCAL_SETTINGS
+  * Store settings in LocalStorage instead of Project Settings Service
+
+* IMJS_MAPBOX_KEY
+  * If defined, sets the MapBox key for the `MapLayerOptions` as an "access_token".
+* IMJS_BING_MAPS_KEY
+  * If defined, sets a Bing Maps key within the `MapLayerOptions` as a "key" type.
+* IMJS_CESIUM_ION_KEY
+  * If defined, the API key supplying access to Cesium ION assets.
+* IMJS_URL_PREFIX
```
Comment (bsteinbk) [inline test-apps/ui-test-app/README.md:69]: This is the only variable that I think would confuse other users as they would not know what to set this to or why they would need to set it. Should you be more specific like it specifies to log into QA DEV or PROD?

Comment (calebmshafer) [reply test-apps/ui-test-app/README.md]: removed that section for now. We can add it back later if needed.

**Pull request merged**